incipyt/actions/setuptools.py

```diff
@@ -56,61 +63,122 @@ def add_to(self, hierarchy):
             "build-backend": "setuptools.build_meta",
         }
 
+        utils.set_items(
+            setup,
+            {
+                "metadata": {
+                    "author_email": "{AUTHOR_NAME} <{AUTHOR_EMAIL}>",
+                    "description": "{SUMMARY_DESCRIPTION}",
+                    "maintainer_email": "{AUTHOR_NAME} <{AUTHOR_EMAIL}>",
+                },
+            },
+            utils.Requires,
+        )
+        utils.set_items(
+            setup,
+            {
+                "metadata": {
+                    "version": "{PACKAGE_VERSION}",
+                },
+                "options": {
+                    "python_requires": ">={PYTHON_VERSION}",
+                },
+            },
+            lambda template: utils.Requires(
+                template,
+                sanitizer=sanitizers.version,
+                PACKAGE_VERSION="0.0.0",
+                PYTHON_VERSION="3.6",
+            ),
+        )
+
         if "metadata" not in setup:
             setup["metadata"] = {}
-        if "options" not in setup:
-            setup["options"] = {}
-        if "options.package_data" not in setup:
-            setup["options.package_data"] = {}
-        if "options.extras_require" not in setup:
-            setup["options.extras_require"] = {}
-
-        utils.append(
+        utils.set_item(
             setup["metadata"],
             "name",
-            utils.Requires("{NAME}", sanitizer=utils.sanitizer_project),
+            utils.Requires("{PROJECT_NAME}", sanitizer=sanitizers.project),
         )
-        utils.append(setup["metadata"], "version", "0.0.1")
-        utils.append(
+
+        if "options" not in setup:
+            setup["options"] = {}
+        utils.set_item(
             setup["options"],
             "packages",
-            utils.Requires("{NAME}", sanitizer=utils.sanitizer_package),
+            utils.Requires("{PROJECT_NAME}", sanitizer=sanitizers.package),
+        )
+
+        if "options.package_data" not in setup:
+            setup["options.package_data"] = {}
+        utils.set_item(
+            setup["options.package_data"],
+            "*",
+            utils.Requires("{PACKAGE_DATA}/*", confirmed=True, PACKAGE_DATA="data"),
         )
-        utils.append(setup["options"], "python_requires", ">=3.6")
-        utils.append(setup["options.package_data"], "*", "data/*")
 
         hierarchy.register_template(
-            Jinja.make("setup.py", sanitizer=utils.sanitizer_package),
+            Jinja.make("LICENSE"),
             Template(
-                """import setuptools
-
-setuptools.setup()
+                """Copyright (c) {{AUTHOR_NAME}}
 
 """
             ),
         )
```

Comment:
Shouldn't license-related stuff have its own action?
This is the kind of thing we're likely to replicate for others build systems

Reply:
Yes I fully agree, but it implies to make our own internal Licence tool, I think it's out of this PR scope.